### DISCUSSION TITLE
Does the CSI provider work on prem?

### DISCUSSION THREAD

**Discussion by Bjego** [Azure/secrets-store-csi-driver-provider-azure#1114] [Q&A] [answered]
Hey guys,
does this CSI provider work in an on prem kubernetes cluster, or is it bound to AKS or Azure Arc?

Kind regards
Bjego

**Comment by nilekhc** [comment, marked as answer]
@Bjego Is your cluster connected via Arc? If yes, you can use a managed solution of Secrets Store CSI driver using the Arc extension.

Otherwise, we can use an open source version of the driver and provider to install in your cluster.
https://azure.github.io/secrets-store-csi-driver-provider-azure/docs/getting-started/installation/#deployment-using-helm

**Reply by Bjego** [reply]
Yes, thank you in the meanwhile I had found the documentation how to use a service principal (which seems to be the only/recommended way for self hosted kubernetes).
https://azure.github.io/secrets-store-csi-driver-provider-azure/docs/configurations/identity-access-modes/service-principal-mode/
